lib/improver/nowcasting/optical_flow.py

```diff
@@ -731,6 +733,40 @@ def calculate_displacement_vectors(self, partial_dx, partial_dy,
 
         return umat, vmat
 
+    @staticmethod
+    def zero_advection_velocities_warning(
+            vel_comp, non_zero_vel_threshold=0.9):
+        """
+        Raise warning if fewer than 90% of the cells within the domain have
+        non-zero advection velocities.
+
+        Args:
+            vel_comp (np.ndarray):
+                Advection velocity that will be checked to assess the
+                proportion of zeroes present in this field.
+
+        Keyword Args:
+            non_zero_vel_threshold (float):
+                Fractional value to specify the proportion of non-zero values
+                that the advection field should contain at a minimum.
+                For example, if non_zero_vel_threshold=0.9 then up to 90% of
```

Comment:
I've inverted this whole method, so that it's dealing with zero advection velocities, rather than non-zero advection velocities, as I think the focus is on the number of zero advection velocities exceeding the defined threshold.